models/models.go

```diff
@@ -20,9 +20,9 @@ import (
 
 	"code.cloudfoundry.org/lager"
 	"github.com/jinzhu/gorm"
+	"github.com/jmcarp/lego/acme"
```

Comment:
I assume this is because `xenolf` became `go-acme`, and `go-acme` has diverged too much to use? (checking because the jmcarp fork hasn't been updated in 8 years, and I happen to know it runs on the deprecated ACME API)

Can/should we take on just ripping the ACME stuff out of the broker, since it doesn't actually do any of it anymore?

Reply:
Longish story

My only intent with this PR was updating the delete server certificate logic in `utils/iam.go` to ignore errors when the certificate was already deleted (since we have a current case of this right now) and adding unit tests  to cover the new behavior

However, this repo is still using `go dep` to manage dependencies, which later versions of Go don't even understand, so it was painful running my tests. So I decided to update from `dep` to Go modules as a side quest.

And in the process, I had to update the code to use `jmcarp/lego` fork of acme, because yeah, it depends on the version of the software as it existed when it was forked.

There is no change materially to the acme source code being used,[ as you can see the code is currently overriding the `lego` package with the `jmcarp/lego` fork](https://github.com/cloud-gov/cf-domain-broker-alb/blob/main/Gopkg.toml#L64-L67)

I hesitated to make any of these packaging changes, but since I don't know how much longer this broker will remain alive, it seemed worthwhile making it easier to operate/test